tests/all/integration/api/speaker/test_speaker_api.py

```diff
@@ -0,0 +1,456 @@
+                    "mobile": "456345678",
+                    "speaking-experience": "Speaking since birth",
+                    "complex-field-values": {
+                        "m.night": "shyamalan"
+                    },  # Should be ignored and saved as None
+                },
+                "relationships": {
+                    "event": {"data": {"id": str(speaker.event_id), "type": "event"}}
+                },
+            }
+        }
+    )
+
+    response = client.post(
+        f'/v1/speakers', content_type='application/vnd.api+json', headers=jwt, data=data,
+    )
+
+    assert response.status_code == 201
+
+    speaker = Speaker.query.get(json.loads(response.data)['data']['id'])
+
+    assert speaker.name == 'Areeb'
+    assert speaker.email == 'jamal.areeb@gmail.com'
+    assert speaker.mobile == '456345678'
+    assert speaker.complex_field_values is None
+
+
+def get_complex_custom_form_speaker(db, user):
+    speaker = get_minimal_speaker(db, user)
+    CustomForms(
+        event=speaker.event,
+        form='speaker',
+        field_identifier='heardFrom',
+        type='text',
+        is_included=True,
+        is_required=True,
+    )
+    CustomForms(
+        event=speaker.event,
+        form='speaker',
+        field_identifier='bestFriend',
+        name='Best Friend',
+        type='text',
+        is_included=True,
+        is_required=True,
+        is_complex=True,
+    )
+    CustomForms(
+        event=speaker.event,
+        form='speaker',
+        field_identifier='transFatContent',
+        name='Trans Fat Content',
+        type='number',
+        is_included=True,
+        is_required=False,
+        is_complex=True,
+    )
+    db.session.commit()
+
+    return speaker
+
+
+def test_custom_form_complex_fields_missing_required(db, client, user, jwt):
+    speaker = get_complex_custom_form_speaker(db, user)
+
+    data = json.dumps(
+        {
+            'data': {
+                'type': 'speaker',
+                'id': str(speaker.id),
+                "attributes": {"name": "Areeb", "email": "jamal.areeb@gmail.com"},
+            }
+        }
+    )
+
+    response = client.patch(
+        f'/v1/speakers/{speaker.id}',
+        content_type='application/vnd.api+json',
+        headers=jwt,
+        data=data,
+    )
+
+    db.session.refresh(speaker)
+
+    assert response.status_code == 422
+    assert json.loads(response.data) == {
+        'errors': [
+            {
+                'detail': "Missing required fields ['best_friend', 'heard_from']",
+                'source': {'pointer': '/data/attributes'},
+                'status': 422,
+                'title': 'Unprocessable Entity',
+            }
+        ],
+        'jsonapi': {'version': '1.0'},
+    }
+
+    assert speaker.name != 'Areeb'
+    assert speaker.email != 'jamal.areeb@gmail.com'
+    assert speaker.complex_field_values is None
+
+
+def test_custom_form_create_complex_fields_missing_required(db, client, user, jwt):
```

Comment:
Codacy found an issue: [Function name "test_custom_form_create_complex_fields_missing_required" doesn't conform to '[a-z_][a-z0-9_]{1,50}$' pattern](https://app.codacy.com/manual/fossasia/open-event-orga-server/pullRequest?prid=5797109)